ChattoAdditions/Source/Input/ContentAwareChatInputBarPresenter.swift

```diff
@@ -0,0 +1,325 @@
+            break
+        }
+        return firstKeyboardInputItem
+    }
+
+    private func cleanCurrentInputView(animated: Bool = false, completion: (() -> Void)? = nil) {
+        self.currentInputView?.endEditing(false)
+        if animated {
+            UIView.animate(withDuration: CATransaction.animationDuration(),
+                           animations: {
+                            self.currentInputView?.alpha = 0.0
+            },
+                           completion: { (_) in
+                            self.currentInputView?.removeFromSuperview()
+                            completion?()
+            })
+        } else {
+            self.currentInputView?.removeFromSuperview()
+            completion?()
+        }
+    }
+
+    private func setupInputView(toContainerController containerController: ContainerControllerProtocol, _ inputView: UIView) {
+        self.shouldIgnoreContainerBottomMarginUpdates = true
+        containerController.changeContainerBottomMargin(withNewValue: self.keyboardHeight, animated: true, callback: {
+            self.shouldIgnoreContainerBottomMarginUpdates = false
+        })
+        let containerView: InputContainerView = {
+            let containerView = InputContainerView()
+            containerView.allowsSelfSizing = true
+            containerView.translatesAutoresizingMaskIntoConstraints = false
+            containerView.contentView = inputView
+            return containerView
+        }()
+        containerController.contentContainer.addSubview(containerView)
+        NSLayoutConstraint.activate([
+            containerView.topAnchor.constraint(equalTo: containerController.contentContainer.topAnchor),
+            containerView.leftAnchor.constraint(equalTo: containerController.contentContainer.leftAnchor),
+            containerView.rightAnchor.constraint(equalTo: containerController.contentContainer.rightAnchor),
+            containerView.bottomAnchor.constraint(equalTo: containerController.contentContainer.bottomAnchor)
+            ])
+        self.currentInputView = containerView
+    }
+
+    private var lastKnownKeyboardHeight: CGFloat?
+    private var keyboardHeight: CGFloat {
+        return self.lastKnownKeyboardHeight ?? self.defaultKeyboardHeight
+    }
+    private var allowListenToChangeFrameEvents = true
+
+    // MARK: Input View
+
+    private weak var currentInputView: InputContainerView?
+
+    private var defaultKeyboardHeight: CGFloat {
+        if UIApplication.shared.statusBarOrientation.isPortrait {
+            return UIScreen.main.defaultPortraitKeyboardHeight
+        } else {
+            return UIScreen.main.defaultLandscapeKeyboardHeight
+        }
+    }
+
+    private func expandedInputViewHeight(forItem item: ChatInputItemProtocol) -> CGFloat {
+        if UIApplication.shared.statusBarOrientation.isPortrait {
+            return UIScreen.main.fixedCoordinateSpace.bounds.height - item.expandedStateTopMargin
```

Comment:
This is wrong. We shouldn't make any assumption about how chat view is used in the app. It may not be displayed in full screen. I was expecting smthg like this `expanded height = chat view height - top margin`